c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1661,34 +1611,104 @@ std::shared_ptr<void> getIpcDevPtr(std::string handle) {
   return sp;
 }
 
-void* raw_alloc(size_t nbytes) {
-  if (nbytes == 0) {
-    return nullptr;
+} // namespace Native
+
+// Define config stuff here, rather than its own .cpp file,
+// because parseArgs needs to know kLargeBuffer.
+bool parsed = false;
+
+size_t m_max_split_size = std::numeric_limits<size_t>::max();
+std::string m_allocator_backend = "native";
+
+void parseArgs() {
+  const char* val = getenv("PYTORCH_CUDA_ALLOC_CONF");
+  if (val != NULL) {
+    const std::string config(val);
+
+    std::regex exp("[\\s,]+");
+    std::sregex_token_iterator it(config.begin(), config.end(), exp, -1);
+    std::sregex_token_iterator end;
+    std::vector<std::string> options(it, end);
+
+    bool used_max_split_size_mb(false);
+    bool used_cudaMallocAsync(false);
+
+    for (auto option : options) {
+      std::regex exp2("[:]+");
+      std::sregex_token_iterator it2(option.begin(), option.end(), exp2, -1);
+      std::sregex_token_iterator end2;
+      std::vector<std::string> kv(it2, end2);
+      if (kv.size() >= 2) {
+        /* Maximum split size in MB.  Limited to large size blocks */
+        if (kv[0].compare("max_split_size_mb") == 0) {
+          size_t val2 = stoi(kv[1]);
+          TORCH_CHECK(
+              val2 > Native::kLargeBuffer / (1024 * 1024),
+              "CachingAllocator option max_split_size_mb too small, must be > ",
+              Native::kLargeBuffer / (1024 * 1024),
+              "");
+          val2 = std::max(val2, Native::kLargeBuffer / (1024 * 1024));
+          val2 = std::min(
+              val2, (std::numeric_limits<size_t>::max() / (1024 * 1024)));
+          m_max_split_size = val2 * 1024 * 1024;
+          used_max_split_size_mb = true;
+        } else if (kv[0].compare("backend") == 0) {
+          TORCH_CHECK(((kv[1].compare("native") == 0) ||
+                       (kv[1].compare("cudaMallocAsync") == 0)),
+                      "Unknown allocator backend, "
+                      "options are native and cudaMallocAsync");
+          m_allocator_backend = kv[1];
+          used_cudaMallocAsync = (kv[1].compare("cudaMallocAsync") == 0);
+          if (used_cudaMallocAsync) {
+#if CUDA_VERSION > 11040
+            int version;
+            C10_CUDA_CHECK(cudaDriverGetVersion(&version));
+            TORCH_CHECK(version >= 11040,
```

Comment:
It looks like this is only checked if `CUDA_VERSION > 11040`, which means if `CUDA_VERSION == 11040`, it won't be checked, right?

I see a number of `CUDA_VERSION > 11040` checks elsewhere, are those meant to be `>` or `>=`?

Reply:
Oops, you're right, i'll change that.